Removed access token and added gradle script

cammace opened the pull request:
Closes #291

zugaldia commented on gradle-config.gradle:
```diff
@@ -0,0 +1,22 @@
+      "    <string name=\"mapbox_access_token\">" + "$System.env.MAPBOX_ACCESS_TOKEN" + "</string>\n" +
+      "</resources>"
+
+    if (tokenFileContents == null) {
```
@cammace @Guardiola31337 This is never true, right? Even if `$System.env.MAPBOX_ACCESS_TOKEN` isn't set, `tokenFileContents` will have some content.

Guardiola31337 replied:
@zugaldia you're completely right.
👀  I modified a bit the script [here](https://github.com/mapbox/mapbox-navigation-android/pull/11).
BTW, we should [get a consensus](https://github.com/mapbox/mapbox-gl-native/issues/8783#issuecomment-296622518) and have same script across all repos.
cc @cammace

zugaldia replied:
@Guardiola31337 great. could you amend this PR with the latest version of the script? Also, instead of failing the build with an `InvalidUserDataException`, would it make sense instead to show a warning and default to a string like `YOUR_MAPBOX_ACCESS_TOKEN_GOES_HERE`?

Guardiola31337 replied:
@zugaldia 
> could you amend this PR with the latest version of the script? 

Sure!

> Also, instead of failing the build with an InvalidUserDataException, would it make sense instead to show a warning and default to a string like YOUR_MAPBOX_ACCESS_TOKEN_GOES_HERE?

That's [what we need to agree on](https://github.com/mapbox/mapbox-gl-native/issues/8783#issuecomment-296622518).
Before, it was implemented that way but instead of generating `YOUR_MAPBOX_ACCESS_TOKEN_GOES_HERE`, `null` was inserted.
It's a matter of _restriction_, because using new script project won't compile until you have `MAPBOX_ACCESS_TOKEN` env var set.
I imagine that this was one of the reasons why it was implemented the way as it is [here](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDKTestApp/gradle-config.gradle), so not being that hard.
That being said, what do you think?

zugaldia replied:
@Guardiola31337 I put my thoughts on https://github.com/mapbox/mapbox-gl-native/issues/8783#issuecomment-297436967.

Guardiola31337 replied:
@cammace [mapbox-navigation-android#29](https://github.com/mapbox/mapbox-navigation-android/pull/29) adds suggestions mentioned.

langsmith commented:
@scothis

cammace commented:
I've rebased this PR and updated the `developer-config.gradle` file. The only thing that still needs to be resolved is that in every example activity we are setting `Mapbox.getInstance(this, getString(R.string.access_token));` but the access_token string doesn't exist. Do we see any reason for keeping this in every file or should we move it out to the application class?

zugaldia commented:
> The only thing that still needs to be resolved is that in every example activity we are setting Mapbox.getInstance(this, getString(R.string.access_token)); but the access_token string doesn't exist. Do we see any reason for keeping this in every file or should we move it out to the application class?

Weren't we doing this because it made it easier to extract the samples into https://www.mapbox.com/android-sdk/examples/ ?

cammace commented:
Yeah, we can keep it if it's beneficial to users, we do document it a bunch now. In that case, we'll need to go through and add a Util of some kind to get the access token.

cammace commented:
Moving forward on this PR, should I close this, create a Util which can load in the access token, or is there a third option? @Guardiola31337 @zugaldia

zugaldia commented:
> Moving forward on this PR, should I close this, create a Util which can load in the access token, or is there a third option?

How about we move [`TokenUtils`](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/utils/TokenUtils.java) to MAS so that we can reuse it in multiple projects?

langsmith commented:
Doing some issue cleanup in this repo.

Move forward on this PR? Close it?

@cammace  I didn't see TokenUtils in MAS, so I'm assuming this hasn't been moved yet...

cc @zugaldia  @Guardiola31337

langsmith commented:
`DO NOT MERGE` label because I still need to make a `Util` or get the access token in some way like @cammace [said above](https://github.com/mapbox/mapbox-android-demo/pull/294#issuecomment-306835918)
